test/benchmark/benchmark_api.cpp

```diff
@@ -0,0 +1,96 @@
+#include <gtest/gtest.h>
+#include <bamit/all.hpp>
+#include <seqan3/core/debug_stream.hpp>
+
+#include <random>
+#include <chrono>
+#include <ctime>
+
+#define RUN(x, y) {startTimeMessage(y);x;endTimeMessage(y);}
+
+time_t _t1, _t2;
+std::chrono::time_point<std::chrono::system_clock> _m1, _m2;
+
+void printTimeMessage(std::string msg)
+{
+    time_t t = time(0);
+    struct tm* now = localtime(&t);
+    seqan3::debug_stream << "[ " << std::put_time(now, "%d-%m-%Y %H:%M:%S") << "] " << msg << '\n';
+}
+
+void startTimeMessage(std::string msg)
+{
+    time(&_t1);
+    _m1 = std::chrono::time_point_cast<std::chrono::milliseconds>(std::chrono::system_clock::now());
+    printTimeMessage("[START] " + msg);
+}
+
+void endTimeMessage(std::string msg)
+{
+    using std::chrono::operator""ms;
+    time(&_t2);
+    _m2 = std::chrono::time_point_cast<std::chrono::milliseconds>(std::chrono::system_clock::now());
+    auto difference = std::chrono::duration_cast<std::chrono::milliseconds>(_m2 - _m1);
+    if (difference < 10000ms)
+        printTimeMessage("[END] " + msg + " (" + std::to_string(difference.count()) + " milliseconds.)");
+    else
+        printTimeMessage("[END] " + msg + " (" + std::to_string((int)difftime(_t2,_t1)) + " seconds.)");
+}
+
+void get_random_position(bamit::Position & start, bamit::Position & end,
+                         seqan3::sam_file_header<std::deque<std::string>> & header)
+{
+    std::random_device rd; // obtain a random number from hardware
+    std::mt19937 gen(rd()); // seed the generator
+    std::uniform_int_distribution<> distr_start(0, header.ref_ids().size() - 1); // define the range
+    uint32_t rand_chr_start = distr_start(gen); // Get start random chromosome.
+    std::uniform_int_distribution<> distr_end(rand_chr_start, header.ref_ids().size() - 1); // define the range
+    uint32_t rand_chr_end = distr_end(gen); // Get end random chromosome, guaranteed to be >= start.
+
+    std::uniform_int_distribution<> distr_pos_start(0, std::get<0>(header.ref_id_info[rand_chr_start]) - 1);
+    uint32_t rand_pos_start = distr_pos_start(gen);
+
+    // Get end position which is greater than or equal to start position.
+    uint32_t end_start = (rand_chr_start == rand_chr_end) ? rand_pos_start : 0;
+    std::uniform_int_distribution<> distr_pos_end(end_start, std::get<0>(header.ref_id_info[rand_chr_end]) - 1);
```

Comment:
What is this end_start? I don't understand the name of the variable.

Reply:
It's basically there so that if start and end chromosomes have been generated to be the same chromosome, the end position cannot start *before* the start position. So imagine rand_chr_start and rand_chr_end both are chr2. Then imagine rand_pos_start is 100. Then, rand_pos_end must be between [100, size(chr2)]. This is to make sure of that.

If they are different chromosomes (rand_chr_start == chr1, rand_chr_end == chr2), then rand_pos_end can obviously be any value from [0, size(chr2)]. So it only applies if they are the same chromosome.